test/core/channelz/channel_trace_test.cc

```diff
@@ -208,11 +208,16 @@ TEST(ChannelTracerTest, ThreadSafety) {
   for (size_t i = 0; i < 10; ++i) {
     threads.push_back(std::make_unique<std::thread>([&]() {
       do {
-        tracer.NewNode("trace").Commit();
+        for (int i = 0; i < 100; i++) {
+          if (done.HasBeenNotified()) return;
+          tracer.NewNode("trace").Commit();
+        }
+        absl::SleepFor(absl::Milliseconds(1));
```

Comment:
`std::this_thread::yield()`?

Reply:
given the lack of guarantees about what that means... and this being working code at the point... eh, happy to review your experiments?